### PR TITLE
Display mail subject in header on mobile

### LINKED ISSUE #5251: Move subject line on mobile into the header bar
## User story
As a mobile user, I have limited space on my screen. That's why I don't want to see a separate line for the subject below the header.

## AC
* the header in single-column mode shows the subject that would be stickied to the top of the view on desktop while scrolling instead of the x/y position (the numbers indicating the mail's position in the list)
* if it's too long for a single line, it should be truncated with ellipses
* the sticky subject is not shown in single-column mode
* the online indicator remains as it is

## Tasks
* [x] Change the `Header` to allow showing text alongside the offline indication 
* [x] Have a callback for the current header from the `ConversationViewer` and adjust the header text in top level view (e.g. `MailView`)
* [x] Remove sticky subject from single-column layout in `ConversationViewer`

## Test Notes
- [ ] Text in the header is displayed as expected in places like the Mail List View, Contacts, Calendar, Settings, etc...
- [ ] When in Mail View, the subject of the mail is displayed
- [ ] If the subject in the mail is long (longer than what can fit in the header) the beginning of the subject is displayed and then cut off
- [ ] Check these on Android (app and mobile firefox/chrome) and ios (app and safari)

### PR DESCRIPTION
**Pull request by wrdhub**
Display mail subject in header on mobile

close #5251

### CODE REVIEW

File: src/mail/view/ConversationViewer.ts [hunk 1]
```diff
@@ -86,7 +87,7 @@ export class ConversationViewer implements Component<ConversationViewerAttrs> {
 				this.renderLoadingState(viewModel),
 				this.renderFooter(),
 			),
-			this.renderFloatingSubject(),
+			client.isMobileDevice() ? null : this.renderFloatingSubject(),
```
Comment (charlag) [inline src/mail/view/ConversationViewer.ts:90]: this should rather depend on single column layout like the rest of the mail layout

Comment (wrdhub) [reply src/mail/view/ConversationViewer.ts:90]: That makes more sense. I had mobile stuck in my head, haha

**Comment by wrdhub** [comment]
> We need to adjust the places where we add the header height so that if we don't display it it's still fine

I thought about this is regards to the scrolling, and I decided not to change there. It will just display some of the previous item, which is something we wanted anyway. Is there anywhere else that we add it that I'm missing?

**Review by charlag — APPROVED** [review]
it's good to go then

**Pull request merged**